File: go/romanTOinteger.go
package main

import (
	"fmt"
	"strings"
)

func romanToInt(s string) int {
	roman_to_int := map[string]int{
		"I": 1,
		"V": 5,
		"X": 10,
		"L": 50,
		"C": 100,
		"D": 500,
		"M": 1000,
	}
	total := 0
	previous_value := 0
	s = strings.ToUpper(strings.TrimSpace(s))

	for _, char := range s {
		value := roman_to_int[string(char)]

		if value > previous_value {
			total += value - 2*previous_value
		} else {
			total += value
		}
		previous_value = value
	}
	return total
}

func main() {
	var input string
	fmt.Print("Type roman number (e.g., III, IV, MCMXCIV): ")
	fmt.Scanln(&input)

	result := romanToInt(input)
	fmt.Printf("The integer value of %s is: %d\n", strings.ToUpper(input), result)
}

File: csharp/romanTOinteger/romanTOinteger.cs
using System;

public class Solution {
    public int RomanValue(char c) {
        switch (c) {
            case 'I': return 1;
            case 'V': return 5;
            case 'X': return 10;
            case 'L': return 50;
            case 'C': return 100;
            case 'D': return 500;
            case 'M': return 1000;
            default: return 0;
        }
    }

    public int RomanToInt(string s) {
        int total = 0;
        int previous_value = 0;
        int value = 0;
        int len = s.Length;

        for (int i = 0; i < len; i++) {
            value = RomanValue(s[i]);
            if (value > previous_value) {
                total += value - 2 * previous_value;
            } else {
                total += value;
            }
            previous_value = value;
        }
        return total;
    }

    static void Main(string[] args) {
        Solution sol = new Solution();
        
        string s1 = "III";
        Console.WriteLine($"Input: {s1}, Output: {sol.RomanToInt(s1)}");
        
        string s2 = "IV";
        Console.WriteLine($"Input: {s2}, Output: {sol.RomanToInt(s2)}");
        
        string s3 = "MCMXCIV";
        Console.WriteLine($"Input: {s3}, Output: {sol.RomanToInt(s3)}");
    }
}